flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/rules/logical/AsyncCalcSplitRule.java

```diff
@@ -71,24 +72,35 @@ public class AsyncCalcSplitRule {
      * org.apache.flink.table.functions.AsyncScalarFunction}.
      */
     public static class AsyncRemoteCalcCallFinder implements RemoteCalcCallFinder {
+
+        private final FunctionKind functionKind;
+
+        public AsyncRemoteCalcCallFinder() {
+            this(FunctionKind.ASYNC_SCALAR);
+        }
```

Comment:
nit: Not sure if we want to default to scalar function given both are supported now. Requiring functionKind is probably fine